zproject/backends.py

```diff
@@ -1843,6 +1843,21 @@ def choose_subdomain(self, relayed_params: Dict[str, Any]) -> Optional[str]:
         else:
             return request_subdomain
 
+    def _check_entitlements(self, idp: SAMLIdentityProvider, attributes: Dict[str, List[str]]) -> None:
+        org_membership_attribute = idp.conf.get('attr_org_membership', None)
+        if org_membership_attribute is None:
+            return
+
+        subdomain = self.strategy.session_get('subdomain')
+        entitlements: Union[str, List[str]] = attributes.get(org_membership_attribute, [])
+        if subdomain not in entitlements:
+            if (subdomain == Realm.SUBDOMAIN_FOR_ROOT_DOMAIN and
```

Comment:
Let's add a comment above this line explaining the purpose here. And I think we also should check `not ROOT_DOMAIN_LANDING_PAGE`

Reply:
Fixed